L/LibGit2/build_tarballs.jl

```diff
@@ -1,7 +1,7 @@
 using BinaryBuilder, Pkg
 
 name = "LibGit2"
-version = v"1.2.2"
+version = v"1.2.3"
```

Comment:
I guess I should have caught this earlier (and not new in this PR), but I guess we're just inventing version numbers here? What happens when upstream actually releases a 1.2.3 release?

Reply:
We are already inventing numbers for LibGit2, upstream is at 1.1.0

Reply:
Yes, I know. Seems like maybe we shouldn't be?

Reply:
Yeah, numbering of JLLs is complicated...  Also other packages diverged from upstream already, we simply don't have enough numbers to distinguish a breaking build within the same upstream version, the only option is to diverge from upstream

Reply:
This was in particular a problem for Julia dependencies, as the new platforms (aarch64-apple-darwin and armv6l) are supported only by Julia v1.6+, so we really could not keep the same version as upstream.  Well, we realised that after wreaking havoc in the ecosystem for a few hours